tools/topology/sof-apl-pcm512x-dual-ssp.m4

```diff
@@ -0,0 +1,214 @@
+
+# playback DAI is SSP5 using 2 periods
+# Buffers use s24le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	1, SSP, 5, SSP5-Codec,
+	PIPELINE_SOURCE_1, 2, s24le,
+	1000, 0, 0, SSP_SCHEDULE_TIME_DOMAIN)
+
+# capture DAI is SSP5 using 2 periods
+# Buffers use s16le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	6, SSP, 5, SSP5-Codec,
+	PIPELINE_SINK_6, 2, s24le,
+	1000, 0, 0, SSP_SCHEDULE_TIME_DOMAIN)
+
+# Media playback pipeline 5 on PCM 4 using max 2 channels of s16le.
+# Set 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-pcm-media.m4,
+	5, 4, 2, s16le,
+	1000, 0, 0,
+	8000, 96000, FSYNC,
+	SSP_SCHEDULE_TIME_DOMAIN,
+	PIPELINE_PLAYBACK_SCHED_COMP_1)
+
+# Connect pipelines together
+SectionGraph."media-pipeline" {
+	index "0"
+
+	lines [
+		dapm(PIPELINE_MIXER_1, PIPELINE_SOURCE_5)
+	]
+}
+
+# playback DAI is iDisp1 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	2, HDA, 0, iDisp1,
+	PIPELINE_SOURCE_2, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp2 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	3, HDA, 1, iDisp2,
+	PIPELINE_SOURCE_3, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp3 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	4, HDA, 2, iDisp3,
+	PIPELINE_SOURCE_4, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# PCM Low Latency, id 0
+dnl PCM_PLAYBACK_ADD(name, pcm_id, playback)
+PCM_DUPLEX_ADD(Port5, 0, PIPELINE_PCM_1, PIPELINE_PCM_6)
+PCM_PLAYBACK_ADD(HDMI1, 5, PIPELINE_PCM_2)
+PCM_PLAYBACK_ADD(HDMI2, 6, PIPELINE_PCM_3)
+PCM_PLAYBACK_ADD(HDMI3, 7, PIPELINE_PCM_4)
+
+#
+# BE configurations - overrides config in ACPI if present
+#
+
+dnl DAI_CONFIG(type, dai_index, link_id, name, ssp_config/dmic_config)
+#SSP 5 (ID: 0)
+ifdef(`CODEC_MASTER',
+`
+DAI_CONFIG(SSP, 5, 0, SSP5-Codec,
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24576000, codec_mclk_in),
+		SSP_CLOCK(bclk, 3072000, codec_master),
+		SSP_CLOCK(fsync, 48000, codec_master),
+		SSP_TDM(2, 32, 3, 3),
+		SSP_CONFIG_DATA(SSP, 5, 24)),
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24576000, codec_mclk_in),
+		SSP_CLOCK(bclk, 512000, codec_master),
+		SSP_CLOCK(fsync, 8000, codec_master),
+		SSP_TDM(2, 32, 3, 3),
+		SSP_CONFIG_DATA(SSP, 6, 24)))
+'
+,
+`
+DAI_CONFIG(SSP, 5, 0, SSP5-Codec,
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24576000, codec_mclk_in),
+		SSP_CLOCK(bclk, 3072000, codec_slave),
+		SSP_CLOCK(fsync, 48000, codec_slave),
+		SSP_TDM(2, 32, 3, 3),
+		SSP_CONFIG_DATA(SSP, 5, 24)),
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24576000, codec_mclk_in),
+		SSP_CLOCK(bclk, 512000, codec_slave),
+		SSP_CLOCK(fsync, 8000, codec_slave),
+		SSP_TDM(2, 32, 3, 3),
+		SSP_CONFIG_DATA(SSP, 6, 24)))
```

Comment:
why a new topology instead of updating the existing one?

Reply:
oh just for testing. Anyway if I do the above suggestion, I probably need to make new file if I don't want to touch existing topologies. But no preferences, if this works reliably, I can also update existing topology.